pkg/selector/outputs/bubbletea.go

```diff
@@ -0,0 +1,227 @@
+// tea model based on user input
+func (m BubbleTeaModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	// check for quit
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch msg.String() {
+		case "ctrl+c", "esc", "q":
+			return m, tea.Quit
+		}
+	case tea.WindowSizeMsg:
+		// handle screen resizing
+
+		// This is needed to handle a bug with bubble tea
+		// where resizing causes misprints (https://github.com/Evertras/bubble-table/issues/121)
+		termenv.ClearScreen()
+
+		// handle width changes
+		m.TableModel = m.TableModel.WithMaxTotalWidth(msg.Width)
+
+		// handle height changes
+		if headerAndFooterPadding >= msg.Height {
+			// height too short to fit rows
+			m.TableModel = m.TableModel.WithPageSize(0)
+		} else {
+			newRowsPerPage := msg.Height - headerAndFooterPadding
+			m.TableModel = m.TableModel.WithPageSize(newRowsPerPage)
+		}
+	}
+
+	// update table
+	var cmd tea.Cmd
+	m.TableModel, cmd = m.TableModel.Update(msg)
+
+	// update footer
+	controlsStr := lipgloss.NewStyle().Faint(true).Render(controlsString)
+	footerStr := fmt.Sprintf("Page: %d/%d | %s", m.TableModel.CurrentPage(), m.TableModel.MaxPages(), controlsStr)
+	m.TableModel = m.TableModel.WithStaticFooter(footerStr)
+
+	return m, cmd
+}
+
+// View is used by bubble tea to render the bubble tea model
+func (m BubbleTeaModel) View() string {
+	outputStr := strings.Builder{}
+
+	outputStr.WriteString(m.TableModel.View())
+	outputStr.WriteString("\n")
+
+	return outputStr.String()
+}
+
+// table creation helpers:
+
+// createRows creates a row for each instance type in the passed in list
+func createRows(instanceTypes []*instancetypes.Details) *[]table.Row {
+	rows := []table.Row{}
+
+	// calculate and fetch all column data from instance types
+	columnsData := getWideColumnsData(instanceTypes)
+
+	// create a row for each instance type
+	for _, data := range columnsData {
+		rowData := table.RowData{}
+
+		// create a new row by iterating through the column data
+		// struct and using struct tags as column keys
+		structType := reflect.TypeOf(*data)
+		structValue := reflect.ValueOf(*data)
+		for i := 0; i < structType.NumField(); i++ {
+			currField := structType.Field(i)
+			columnName := currField.Tag.Get(columnTag)
+			colValue := structValue.Field(i)
+			rowData[columnName] = getUnderlyingValue(colValue)
+		}
+
+		newRow := table.NewRow(rowData)
+
+		rows = append(rows, newRow)
+	}
+
+	return &rows
+}
+
+// createColumns creates columns based on the column key constants
+func createColumns() *[]table.Column {
+	columns := []table.Column{}
+
+	// iterate through wideColumnsData struct and create a new column for each field tag
+	columnDataStruct := wideColumnsData{}
+	structType := reflect.TypeOf(columnDataStruct)
+	for i := 0; i < structType.NumField(); i++ {
+		columnHeader := structType.Field(i).Tag.Get(columnTag)
+		newCol := table.NewColumn(columnHeader, columnHeader, len(columnHeader)+headerPadding)
```

Comment:
I noticed that some columns are sized too small for the data (like CPU Arch and GPU Info). You should be able to dynamically calculate the columns header by looking for the max(row entry per column, column header, and some upper bound of column length).